cmd/falconer/main.go

```diff
@@ -42,6 +46,31 @@ func main() {
 		log.WithError(err).Fatal("Failed to listen")
 	}
 	var opts []grpc.ServerOption
+	if len(config.TLS.Certificates) > 0 {
+		tlsConfig := &tls.Config{
+			Certificates: []tls.Certificate{},
+			ClientCAs:    x509.NewCertPool(),
+		}
+		for _, certConfig := range config.TLS.Certificates {
+			cert, err := tls.LoadX509KeyPair(certConfig.Cert, certConfig.Key)
+			if err != nil {
+				log.WithError(err).Fatal("Failed to load TLS certificate")
+			}
+			tlsConfig.Certificates = append(tlsConfig.Certificates, cert)
+		}
+
+		for _, caConfig := range config.TLS.ClientCas {
+			caData, err := ioutil.ReadFile(caConfig)
+			if err != nil {
+				log.WithError(err).Fatal("Failed to read client CA")
+			}
+			if !tlsConfig.ClientCAs.AppendCertsFromPEM(caData) {
+				log.Fatal("Failed to load client CA")
+			}
+			tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
```

Comment:
Seems like doing this unconditionally might be more than we want? Maybe we can add configurability around it later, if needed.

Reply:
i'll add configuration for it